### ISSUE TITLE
Support reverse time integration with ODE.

### ISSUE THREAD

**Issue by ibraheem-moosa** [google/jax#7269]
We are trying to implement the Neural ODE paper as part of the [JAX/FLAX Community Week at Hugging Face](https://discuss.huggingface.co/t/open-to-the-community-community-week-using-jax-flax-for-nlp-cv/7104). While trying to implement the Continuous Normalizing Flow part of the paper we realized we have to run the ODE solver reversed in time. 

This feature is not supported by the current `odeint` in [jax.experimental.ode](https://github.com/google/jax/blob/208dd1ac3f5c4835bc3e3986427ec703e5699e73/jax/experimental/ode.py#L153). In contrast this feature is supported in [scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.odeint.html) and [torchdiffeq](https://github.com/rtqichen/torchdiffeq).

It would be great if this feature is supported in the `jax.experimental.ode`.

**Comment by mattjj** [comment]
Good idea!

You might be able to use the same trick used in the vjp rule, where to integrate xdot = f(x) from 0 to -1, you instead integrate [xdot = -f(-x)](https://github.com/google/jax/blob/208dd1ac3f5c4835bc3e3986427ec703e5699e73/jax/experimental/ode.py#L229) from [0 to 1](https://github.com/google/jax/blob/208dd1ac3f5c4835bc3e3986427ec703e5699e73/jax/experimental/ode.py#L245). I think that's essentially what scipy.odeint does, which is why [the time values must be monotonically increasing or decreasing](https://github.com/scipy/scipy/blob/44e31b1eb202ef91caad059952ba268d2610a60b/scipy/integrate/odepack.py#L235-L237).

**Comment by mattjj** [comment]
More concretely, I think `odeint(f, x0, t)`, where t has monotonically decreasing values, would produce the same values as `odeint(lambda x, t: -f(x, -t), x0, -t)`, where in this call the time values are monotonically increasing. I didn't work out the change-of-variable argument, so that claim may be buggy, but something like that should work.

**Comment by jatentaki** [comment]
@ibraheem-moosa is your code available somewhere? I was recently trying to implement CNFs as well and ran into #7142. Are you seeing any issues there?

**Comment by ibraheem-moosa** [comment]
@jatentaki You can find my code [here](https://github.com/ibraheem-moosa/neural-ode-in-jax/blob/master/examples/cnf.py) and [here](https://github.com/ibraheem-moosa/neural-ode-in-jax/blob/master/examples/cnf_jax.py). It is really a WIP. I'm not sure how  helpful it would be. I have been struggling with this for the past few days.

**Comment by sw32-seo** [comment]
@jatentaki Does Flax have the same interaction issue with odeint as haiku does? 
I think I have a similar problem in [here](https://github.com/sw32-seo/neuralODE/blob/CNF/train_cnf.py)
I use -f(x, -t) with t =[10, 0] as @mattjj mentioned. Then, I compared trace at [line 85](https://github.com/sw32-seo/neuralODE/blob/1dff4e38c5c8827cf74e9a81d954d825ea72a99c/train_cnf.py#L85) to [original code](https://github.com/rtqichen/torchdiffeq/blob/master/examples/cnf.py) and confirmed match. 
However, odeint give a very different behavior to the original one.

**Comment by mandelbrot-walker** [comment]
@jatentaki you can find a working CNF [here](https://huggingface.co/flax-community/NeuralODE_SDE) by @sw32-seo .
